src/Sulu/Component/Content/Document/SynchronizationManager.php

```diff
@@ -0,0 +1,235 @@
+     * the operation has completed.
+     *
+     * @param SynchronizeBehavior $document
+     */
+    public function synchronizeFull(SynchronizeBehavior $document, array $options = [])
+    {
+        // get the default managerj
+        $defaultManager = $this->registry->getManager();
+        $publishManager = $this->getPublishDocumentManager();
+
+        // if the  emitting manager is the publish manager, stop we
+        // don't want to sync from the publish workspace!
+        if ($publishManager === $defaultManager) {
+            return;
+        }
+
+        // Get the routes for the document if it implements
+        // the "resource segment subscriber".
+        //
+        // TODO: We should instead add configuration to cascade certain
+        //       document classes instead of hard coding this logic.
+        $routes = [];
+        if ($document instanceof ResourceSegmentBehavior) {
+            $routes = $this->getDocumentRoutes($defaultManager->getInspector(), $document);
+        }
+
+        $toSynchronize = array_merge([
+            $document,
+        ], $routes);
+
+        foreach ($toSynchronize as $syncDocument) {
+            $this->synchronizeSingle($syncDocument, $options);
+        }
+
+        $publishManager->flush();
+        $defaultManager->flush();
+    }
+
+    /**
+     * Synchronize a single document to the publish document manager in the
+     * documents currently registered locale.
+     *
+     * FLUSH will not be called and no associated documents will be
+     * synchronized.
+     *
+     * TODO: Add an explicit "locale" option?
+     *
+     * @param SynchronizeBehavior $document
+     * @param bool $force
+     */
+    public function synchronizeSingle(SynchronizeBehavior $document, array $options = [])
+    {
+        $options = array_merge([
+            'force' => false,
+        ], $options);
+
+        $defaultManager = $this->registry->getManager();
+        $publishManager = $this->getPublishDocumentManager();
+
+        // see comment in same condition above.
+        if ($publishManager === $defaultManager) {
+            return;
+        }
+
+        // get the list of managers with which the document is already
+        // synchronized (the value may be NULL as we have no control over what
+        // the user does with this mapped value).
+        $synced = $document->getSynchronizedManagers() ?: [];
+
+        // unless forced, we will not process documents which are already
+        // synced with the publish document manager.
+        if (false === $options['force'] && in_array($this->publishManagerName, $synced)) {
+            return;
+        }
+
+        $inspector = $defaultManager->getInspector();
+        $locale = $inspector->getLocale($document);
+        $path = $inspector->getPath($document);
+
+        // register the DDM document and its immediate relations with the PDM
+        // PHPCR node.
+        $this->registrator->registerDocumentWithPDM($document);
```

Comment:
Should be `registerNodeWithTargetManager` or similar, and the name of the target manager should be provided.